manifests/grafana.pp

```diff
@@ -45,6 +45,45 @@
     manage_package_repo => false,
     version             => $puppet_metrics_dashboard::grafana_version,
     cfg                 => $grafana_cfg,
-    require             => Service[$puppet_metrics_dashboard::influx_db_service_name],
+    require             => Http_conn_validator['influxdb-conn-validator'],
+    notify              => Exec['update Grafana admin password'],
   }
+
+  $_uri = $puppet_metrics_dashboard::use_dashboard_ssl ? {
+    true    => 'https',
+    default => 'http',
+  }
+
+  exec { 'update Grafana admin password':
+    path        => '/usr/bin',
+    command     => @("CHANGE_GRAFANA_PW"),
```

Comment:
Still seeing issues with this on the initial run
```
Error: /Stage[main]/Puppet_metrics_dashboard::Grafana/Exec[update Grafana admin password]: Failed to call refresh: 'curl -X PUT -H "Content-Type: application/json" -d '{
  "oldPassword": "admin",
  "newPassword": "admin",
  "confirmNew": "admin"
}' http://admin:admin@localhost:8080/api/user/password
' returned 7 instead of one of [0]
```
and attempting to change the password:
```
Error: /Stage[main]/Puppet_metrics_dashboard::Post_start_configs/Grafana_datasource[influxdb_telegraf]: Could not evaluate: Fail to retrieve datasources (HTTP response: 401/{"message":"Invalid username or password"})
```

Reply:
What's your test methodology here? I ask cause, as shown in the output above, a single pass works fine on CentOS 7.

Reply:
This is a vmpooler host (Debian 9) running against a 2018.1.5 master.  I did notice that the compiled catalog looks a little odd wrt the exec resource in the `puppet_metrics_dashboard::grafana` class:

```
            "parameters": {
                "command": "curl -X PUT -H \"Content-Type: application/json\" -d '{\n  \"oldPassword\": \"admin\",\n  \"newPassword\": \"password\",\n  \"confirmNew\": \"password\"\n}' http://admin:admin@localhost:3000/api/user/password\n",
                "cwd": "/usr/share/grafana",
                "path": "/usr/bin",
                "refreshonly": true
            },
```

Maybe the line breaks are the problem?

Reply:
Shouldn't be but at least now I've got an idea on what to look at.